### ISSUE TITLE
Dropout2d doesn't drop channels for (C, H, W)

### ISSUE THREAD

**Issue by OverLordGoldDragon** [pytorch/pytorch#69801]
### 🐛 Describe the bug

[Docs](https://pytorch.org/docs/1.10.0/generated/torch.nn.Dropout2d.html?highlight=dropout2d#torch.nn.Dropout2d) read that `C` is dropped, which does not occur for `(C, H, W)` 

```python
import torch
torch.manual_seed(0)
ipt = torch.ones((2, 3, 4))
print(torch.nn.Dropout2d(p=0.5)(ipt))
```
```
tensor([[[2., 2., 2., 2.],
         [0., 0., 0., 0.],
         [0., 0., 0., 0.]],
        [[2., 2., 2., 2.],
         [2., 2., 2., 2.],
         [0., 0., 0., 0.]]])
```
but does for `(N, C, H, W)`
```python
torch.manual_seed(0)
print(torch.nn.Dropout2d(p=0.5)(ipt[None]))
```
```
tensor([[[[2., 2., 2., 2.],
          [2., 2., 2., 2.],
          [2., 2., 2., 2.]],
         [[0., 0., 0., 0.],
          [0., 0., 0., 0.],
          [0., 0., 0., 0.]]]])
```
It appears `Dropout2d` is instead implementing `Dropout1d` for `(N, C, T)`.

### Versions

1.10.0 via Anaconda; Windows 10

cc @ezyang @gchanan @zou3519 @bdhirsh @albanD @mruberry @jbschlosser @walterddr @kshitij12345

**Comment by kshitij12345** [comment]
Even Dropout3d is affected with the same issue. Seems like the underlying implementation always assumes that the input will be batched.

https://github.com/pytorch/pytorch/blob/fc37e5b3ed0e51a0dff39da6e3a1a3ce16fe2756/aten/src/ATen/native/Dropout.cpp#L19-L20

**Comment by albanD** [comment]
cc @jbschlosser should this be fixed when no batch dim support is added?

**Comment by kshitij12345** [comment]
FYI, as per the tracker issue https://github.com/pytorch/pytorch/issues/60585 it is already supported.

**Comment by albanD** [comment]
Ok, so most likely a bug then!

**Comment by jbschlosser** [comment]
Reopening this thanks to #79549

TL;DR: The fix for this issue was silently BC-breaking for those who depended on `Dropout2d` to do 1D channel-wise dropout for 3D inputs. For some reason, `Dropout1d` didn't exist before, so `Dropout2d` was the only option. We'll revert to the old behavior for 1.12 with a warning to move to the newly-added `Dropout1d`. In a release, we'll switch back to a no-batch-dim interpretation for 3D inputs.
